pkg/operator2/oauth.go

```diff
@@ -136,3 +148,173 @@ func (c *osinOperator) handleOAuthConfig(configOverrides []byte) (*corev1.Config
+				},
+			},
+			Organizations: githubConfig.Organizations,
+			Hostname:      githubConfig.Hostname,
+			CA:            getLocalConfigMapFilename(githubConfig.CA),
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeGitLab:
+		gitlabConfig := providerConfig.GitLab
+		p := osinv1.GitLabIdentityProvider{
+			CA:       getLocalConfigMapFilename(gitlabConfig.CA),
+			URL:      gitlabConfig.URL,
+			ClientID: gitlabConfig.ClientID,
+			ClientSecret: configv1.StringSource{
+				// TODO: this could be Env, Value, File, either can be encrypted
+				StringSourceSpec: configv1.StringSourceSpec{
+					File: getLocalSecretFilename(gitlabConfig.ClientSecret),
+				},
+			},
+			Legacy: &gitlabConfig.LegacyOAuth2, // TODO: better mapping of 2 to 3 values?
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeGoogle:
+		googleConfig := providerConfig.Google
+		p := osinv1.GoogleIdentityProvider{
+			ClientID: googleConfig.ClientID,
+			ClientSecret: configv1.StringSource{
+				// TODO: this could be Env, Value, File, either can be encrypted
+				StringSourceSpec: configv1.StringSourceSpec{
+					File: getLocalSecretFilename(googleConfig.ClientSecret),
+				},
+			},
+			HostedDomain: googleConfig.HostedDomain,
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeOpenID:
+		openIDConfig := providerConfig.OpenID
+		p := osinv1.OpenIDIdentityProvider{
+			CA:       getLocalConfigMapFilename(openIDConfig.CA),
+			ClientID: openIDConfig.ClientID,
+			ClientSecret: configv1.StringSource{
+				// TODO: this could be Env, Value, File, either can be encrypted
+				StringSourceSpec: configv1.StringSourceSpec{
+					File: getLocalSecretFilename(openIDConfig.ClientSecret),
+				},
+			},
+			ExtraScopes:              openIDConfig.ExtraScopes,
+			ExtraAuthorizeParameters: openIDConfig.ExtraAuthorizeParameters,
+			URLs: osinv1.OpenIDURLs{ // TODO: do we need this to be different from configv1?
+				Authorize: openIDConfig.URLs.Authorize,
+				Token:     openIDConfig.URLs.Token,
+				UserInfo:  openIDConfig.URLs.UserInfo,
+			},
+			Claims: osinv1.OpenIDClaims{ // TODO: do we need this to be different from configv1?
+				ID:                []string{"sub"}, // TODO: this is missing in configv1 version?
+				PreferredUsername: openIDConfig.Claims.PreferredUsername,
+				Name:              openIDConfig.Claims.Name,
+				Email:             openIDConfig.Claims.Email,
+			},
+		}
+		serializer.Encode(&p, bytesWriter)
+	} // switch
+
+	bytesWriter.Flush()
+	return providerConfigBytes.Bytes()
+}
+
+// TODO: move to some kind of helpers file, possibly branch to Local/Remote NS versions if necessary
+// TODO: the logic of naming the CMs and Secrets to be mounted to the OSIN container
+// TODO: we need to keep track of everything that needs to be mounted in the containers
+func getLocalConfigMapFilename(ref configv1.ConfigMapReference) string {
```

Comment:
Both of these helper functions should return a string:

`/var/config/user/idp/<index_of_idp>/<configmap_or_secret>/<name_of_key>`

I believe that is enough for us to generally identify everything.  We would mount it at `/var/config/user/idp/<index_of_idp>/<configmap_or_secret>`.

I could see some argument for replacing `<index_of_idp>` with `<name_of_idp>`.  Either way, enough to get started.

As a test, I would like to see what the generated config looks like.

Reply:
Hmm actually it needs the name of the config map or secret as well since they could contain conflicting keys.

Reply:
I thought we would rather have a more general map of (config_map, key) -> (filename) which we supply to the method taking care of creating the Deployment object so that this one can use it to add proper mounting mapping.

I think the `name/index_of_idp` might be irrelevant. All you want is your config maps to be mounted into right filenames, irregardless of their purpose?